### PR TITLE
Unique Flaregun Flare

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Uses the Mortar flare.
Only flaregun can fire it.
Tye told me to change these to static lights so they're static lights now.

https://user-images.githubusercontent.com/46353991/202874135-d7bfaff2-08bd-4f1a-8cc3-f2f1b126a27e.mp4
## Why It's Good For The Game
Gives flare gun more utility, and makes sense that it would have something like it to me.
## Changelog
:cl:
add: Added new flaregun flare type, works exactly like mortar flare.
fix: Overhead 
/:cl:

### CODE REVIEW

**Comment by iLikeMemes** [comment]
As a flare enthousiast I really like this, However I do think it **_might_** be too strong for it to be an unlimited item in the marine vendor. I cant really say for sure as i am not 100% sure how long those flares last. 
other than that,
**speed merge**

**Comment by DemonOfHatred** [comment]
Theres already so many things that give off light, the endless revolver spam for one. not needed.

**Comment by Lumipharon** [comment]
This needs to significantly weaker (probably durationwise) than cas/mortar flares.

**Comment by Losenis** [comment]
Does it really? It already sacrifices the funny "set xeno on fire if hit" thing that you can shove into underbarrel attachments, even already triggered flares. And normal flares you can grab and throw further away once you need light further ahead, not so much with these.

**Comment by FosterGrime-Heymynameslouie** [comment]
how about give flare guns iff instead of a stronger flare
I think it would give flare guns more utility

cm does it and it's a huge qol iirc

**Comment by novaepee** [comment]
> how about give flare guns iff instead of a stronger flare I think it would give flare guns more utility
> 
> cm does it and it's a huge qol iirc

In code, a flare is a grenade not a bullet, so you have to make it IFF worthy.

**Comment by FosterGrime-Heymynameslouie** [comment]
> > how about give flare guns iff instead of a stronger flare I think it would give flare guns more utility
> > cm does it and it's a huge qol iirc
> 
> In code, a flare is a grenade not a bullet, so you have to make grenade IFF which is not an easy feat.

oh

**Comment by Al-1ce** [comment]
I like this very much. Duration shouldn't be nerfed to the point that this becomes obsolete, though.

**Comment by FIXTHERADBOXPLEASE** [comment]
Great. A PR that found a way to powercreep flares, of all things. Not just that, but one that removes all xeno counterplay to them.

**Comment by zzurge** [comment]
LET THERE BE LIGHT

**Comment by MereWalker** [comment]
Should make it last for 12 seconds and it then fizzles out. You could also make it flicker it's light so doesn't bright light all the time. Just a thought.

**Comment by Araulius** [comment]
My dude decided to make a special flare for the flare gun, but not a special cas flare for the flare gun

**Comment by Overdrive659** [comment]
This is cool as fuck, however;

Consider either-
limiting flare gun for SLs, FCs and other command
limiting flare guns + ammo in vendors

Currently, this makes the flare gun VERY powerful: It has no xeno counterplay, as xenos cannot apply acid to it, and it is a lot brighter than normal flares.
When CAS uses its flares it feels balanced, having that much light for that duration. Making that handheld and infinate.... IDK man.

**Comment by Al-1ce** [comment]
I think making this an engi item would be overall better and also make sense. Also maybe as a selectable pouch in the loadout vendors with a limited amount of ammo.

**Comment by Al-1ce** [comment]
I think this would be perfect as an engineer item.

File: code/game/objects/items/explosives/grenades/marines.dm [hunk 1]
```diff
@@ -462,3 +462,23 @@
...
+	item_state = "grenade_smoke"
+	det_time = 0
+	dangerous = FALSE
+	hud_state = "grenade_smoke"
+	dangerous = FALSE
+	icon_state_mini = "grenade_blue"
+
+/obj/item/explosive/grenade/flaregun/attack_self(mob/user)
+	to_chat(user, span_warning("You can't use this without a flaregun!"))
+	return
```
Comment (TiviPlus) [inline code/game/objects/items/explosives/grenades/marines.dm:479]: why is thzis a grenade attack self should be calling parent

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/items/explosives/grenades/marines.dm [hunk 2]
```diff
@@ -462,3 +462,23 @@
 	else
 		icon_state = initial(icon_state)
 		set_light(0)
+
+/obj/item/explosive/grenade/flaregun
+	name = "\improper M60 FLHA Flare"
```
Comment (XSlayer300) [inline code/game/objects/items/explosives/grenades/marines.dm:467]:
```suggestion
	name = "\improper M60 FLHA flare shell"
```

**Comment by Starfirejordan** [comment]
causing an issue that makes CAS flares not work
image

**Comment by Pariah919** [comment]
too tired to finish this off, I'll come back to this in a while.

**Pull request closed without merging**